analyzers/src/SonarAnalyzer.CSharp/Rules/MultilineBlocksWithoutBrace.cs

```diff
@@ -28,143 +28,125 @@
-            //    currentIfStatement = statement as IfStatementSyntax;
-            //}
-
-            //return statement;
+            statement = ifStatement.Else.Statement;
+            ifStatement = statement as IfStatementSyntax;
         }
+        return statement;
+    }
 
-        private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax statement,
-            string executed, string execute)
+    private static void CheckStatement(SyntaxNodeAnalysisContext context, StatementSyntax first, string executed, string execute)
+    {
+        if (SecondStatement(context.Node, first) is { } second && IsNotEmpty(first) && IsNotEmpty(second))
         {
-            if (statement.IsKind(SyntaxKind.Block))
-            {
-                return;
-            }
+            var firstPosition = ProxyStartPosition(first);
+            var secondPosition = StartPosition(second);
 
-            var nextStatement = context.Node.GetLastToken().GetNextToken().Parent;
-            // This algorithm to get the next statement can sometimes return a parent statement (for example a BlockSyntax)
-            // so we need to filter this case by returning if the nextStatement happens to be one ancestor of statement.
-            if (nextStatement == null ||
-                statement.Ancestors().Contains(nextStatement))
+            if (secondPosition.Character > firstPosition.Character - IndentSize)
             {
-                return;
+                var lineSpan = context.Node.SyntaxTree.GetText().Lines[secondPosition.Line].Span;
+                var location = Location.Create(context.Node.SyntaxTree, TextSpan.FromBounds(second.SpanStart, lineSpan.End));
+                var blockSize = secondPosition.Line - StartPosition(first).Line + 1;
+                var additional = new[] { first.GetLocation() };
+                context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, location, additional, executed, execute, blockSize));
             }
+        }
+    }
 
-            var statementPosition = statement.GetLocation().GetLineSpan().StartLinePosition;
-            var nextStatementPosition = nextStatement.GetLocation().GetLineSpan().StartLinePosition;
+    private static bool IsNotEmpty(SyntaxNode node) =>
+        node is not EmptyStatementSyntax;
 
-            if (statementPosition.Character == nextStatementPosition.Character)
-            {
-                var lineSpan = context.Node.SyntaxTree.GetText().Lines[nextStatementPosition.Line].Span;
-                var location = Location.Create(context.Node.SyntaxTree, TextSpan.FromBounds(nextStatement.SpanStart, lineSpan.End));
+    private static LinePosition ProxyStartPosition(SyntaxNode node)
+    {
+        var line = StartPosition(node).Line;
+        return node.AncestorsAndSelf()
+            .Where(x => StartPosition(x).Line == line)
+            .Select(x => Proxy(x, x != node))
+            .Last();
```

Comment:
So it can still be like this, to settle the cardinaltity down, right?
```suggestion
        var firstOnLine = node.AncestorsAndSelf().Last(x => StartPosition(x).Line == line);
        return Proxy(firstOnLine, firstOnLine != node);
```
Having that, we can remove `Proxy` method and just do the action directly.